checker.cpp

```diff
@@ -1,22 +1,294 @@
+    {
+        if (!strcmp(lang, "English"))
+        {
+            cout << "Warning: Approaching charge-peak\n";
+        }
+        if (!strcmp(lang, "German"))
+        {
+            cout << "Warnung: Ladespitze nähert sich\n";
+        }
+    }
+
+    void LogLowSocWarning(char* lang)
+    {
+        if (!strcmp(lang, "English"))
+        {
+            cout << "Warning: Approaching discharge\n";
+        }
+        if (!strcmp(lang, "German"))
+        {
+            cout << "Warnung:  Naht Entladung\n";
+        }
+    }
+
+    void LogChargeRateWarning(char* lang)
+    {
+        if (!strcmp(lang, "English"))
+        {
+            cout << "Warning: ChargeRate is high\n ";
+        }
+        if (!strcmp(lang, "German"))
+        {
+            cout << "Warnung:  Naht Entladung\n";
+        }
+    }
+
+    int valuesIsInWarningRange(float value, float min, float max, float minwarn, float maxwarn)
+    {
+      
+        if(lowLimitWarningCheck(value,min,minwarn))
+        {
+            return LOW_LIMIT_WARNING;
+        }
+        if(highLimitWarningCheck(value,max,maxwarn))
+        {
+            return HIGH_LIMIT_WARNING;
+        }
+    }
+    
+    bool lowLimitWarningCheck(float value, float min, float minwarn)
+    {
+        if(min<value<=minwarn)
+        {
+            return true;
+        }
+    }
+    
+    bool highLimitWarningCheck(float value, float max, float maxwarn)
+    {
+        if(maxwarn<=value<max)
+        {
+            return true;
+        }
+    }
+};
+
+void checkBattery()
+{
+    NewBattery n;
+
+    //Check if battery is ok
+    assert(n.checkBatteryIsOk(0, 19, 1) == false);
+    assert(n.checkBatteryIsOk(1, 21, 1) == false);
+    assert(n.checkBatteryIsOk(1, 81, 0.7) == false);
+    assert(n.checkBatteryIsOk(-1, 79, 0.7) == false);
+    assert(n.checkBatteryIsOk(1, 21, 0.7) == true);
+    assert(n.checkBatteryIsOk(44, 79, 0.7) == true);
+
+    //Check if temperature in range
+    assert(n.checkTempIsInRange(0, mintemp, maxtemp) == true);
+    assert(n.checkTempIsInRange(45, mintemp, maxtemp) == true);
+    assert(n.checkTempIsInRange(-1, mintemp, maxtemp) == false);
+    assert(n.checkTempIsInRange(46, mintemp, maxtemp) == false);
+    assert(n.checkTempIsInRange(40, mintemp, maxtemp) == true);
+
+    //Check if soc in range
+    assert(n.checkSocIsInRange(20, minsoc, maxsoc) == true);
+    assert(n.checkSocIsInRange(80, minsoc, maxsoc) == true);
+    assert(n.checkSocIsInRange(19, minsoc, maxsoc) == false);
+    assert(n.checkSocIsInRange(81, minsoc, maxsoc) == false);
+    assert(n.checkSocIsInRange(21, minsoc, maxsoc) == true);
+
+    //Check if chargerate in range
+    assert(n.checkChargeRateIsInRange(0.8) == false);
+    assert(n.checkChargeRateIsInRange(0.9) == false);
+    assert(n.checkChargeRateIsInRange(0.7) == true);
+    assert(n.checkChargeRateIsInRange(0) == true);
+
+    //check temp warning levels
+    assert(n.checkTempIsInWarningRange(1, "English") == true);
```

Comment:
checkTempIsInWarningRange() is not part of checkBatteryIsOk() flow. It should be part of normal flow. If not then it will be just a method used only in test code. Please check if you can update the flow.